packages/SwingSet/src/kernel/liveSlots.js

```diff
@@ -107,6 +107,8 @@ function build(
   const pendingPromises = new Set(); // Promises
   const importedDevices = new Set(); // device nodes
   const deadSet = new Set(); // vrefs that are finalized but not yet reported
+  const possiblyDeadSet = new Set(); // vrefs that might need to be rechecked for being dead
```

Comment:
> On one hand, if the only pillar keeping a VO alive is virtualized data, and userspace explicltly deletes it, that should allow the `syscall.retireExport` to happen immediately, rather than waiting for end-of-crank or `bringOutYourDead`. On the other hand, the probe to see whether the Presence pillar is up or down must query the WeakRef, and those results are _not_ safe to use until `bringOutYourDead` (because organic GC could cause variance). So maybe the safest thing to do _is_ to only allow VOs to be deleted during `processDeadSet`, even if it wasn't a Representative being deleted that enabled it, which means sticking with the pattern you established in this PR.

I'm probably out of my depth here, but why can't the check for the Presence pillar rely on a flag "there is a known Presence, it may be dead, but since we haven't reaped it yet, we don't know", without checking for the WeakRef directly.
So there's be 3 states for presences:
- Live presence (or unreachable presence that hasn't been actively collected by the engine yet)
- Dead presence, but not yet reaped. The finalizer ran, and added it to the dead set
- Dead presence, reaped. BOYD was called.

If userspace deletes virtualized data, it would only immediately retire the export if the presence was reaped. If it wasn't reaped, the export retire would happen when BOYD is called.